src/redecomp/transfer/SparseMatrixTransfer.cpp

```diff
@@ -0,0 +1,306 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Tribol Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (MIT)
+
+#include "SparseMatrixTransfer.hpp"
+
+#include <map>
+
+#include "axom/slic.hpp"
+
+#include "redecomp/utils/MPIUtility.hpp"
+#include "redecomp/utils/MPIArray.hpp"
+#include "redecomp/RedecompMesh.hpp"
+
+namespace redecomp
+{
+
+SparseMatrixTransfer::SparseMatrixTransfer(
+  const mfem::ParFiniteElementSpace& parent_test_space,
+  const mfem::ParFiniteElementSpace& parent_trial_space,
+  const mfem::FiniteElementSpace& redecomp_test_space,
+  const mfem::FiniteElementSpace& redecomp_trial_space
+)
+: parent_test_space_ { parent_test_space },
+  parent_trial_space_ { parent_trial_space },
+  redecomp_test_space_ { redecomp_test_space },
+  redecomp_trial_space_ { redecomp_trial_space },
+  redecomp_test_mesh_ { getRedecompMesh(redecomp_test_space_) },
+  redecomp_trial_mesh_ { getRedecompMesh(redecomp_trial_space_) }
+{
+  SLIC_ERROR_ROOT_IF(static_cast<const mfem::Mesh*>(&redecomp_test_mesh_.getParent()) != parent_test_space_.GetMesh(),
+    "The parent test quadrature space mesh must be linked to the test Redecomp mesh.");
+  SLIC_ERROR_ROOT_IF(static_cast<const mfem::Mesh*>(&redecomp_trial_mesh_.getParent()) != parent_trial_space_.GetMesh(),
+    "The parent trial quadrature space mesh must be linked to the trial Redecomp mesh.");
+  SLIC_ERROR_ROOT_IF(&redecomp_test_mesh_.getMPIUtility().MPIComm() != &redecomp_trial_mesh_.getMPIUtility().MPIComm(),
+    "MPI Communicator must match in test and trial spaces.");
+
+  // one value per element for initial implementation
+  SLIC_ERROR_ROOT_IF(!parent_test_space_.IsDGSpace() || !parent_trial_space_.IsDGSpace() ||
```

Comment:
@kswartz92 , is there a need to make this more general than L2 order 0 in the near term?

Reply:
I don't think so, L2 order 0 will cover most use cases.